src/main/java/redis/clients/util/JedisClusterHashTagUtil.java

```diff
@@ -0,0 +1,41 @@
+package redis.clients.util;
+
+/**
+ * Holds various methods/utilities to manipualte and parse redis hash-tags.
+ * See <a
+ * href="http://redis.io/topics/cluster-spec">Cluster-Spec : Keys hash tags</a>
+ */
+public final class JedisClusterHashTagUtil {
+
+  private JedisClusterHashTagUtil() {
+    throw new InstantiationError("Must not instantiate this class");
+  }
+
+  public static String getHashTag(String key) {
+    return extractHashTag(key, false, false);
+  }
+
+  public static boolean isClusterCompliantMatchPattern(String matchPattern) {
+    String tag = extractHashTag(matchPattern, true, true);
+    return tag != null && !tag.isEmpty() ;
+  }
+
+  private static String extractHashTag(String key, boolean includeBrackets, boolean returnNullWhenTagNotFound) {
```

Comment:
@marcosnils I do agree with the above paradigm when it comes to exposing external methods. Notice that I wrapped the private flag method with 2 non-flag methods just as expressed in the artical . When it comes to internal/private methods with a limited use ( especially in static context ) I feel it's a better approach as the alternative would be duplicating the code.

Anyway , can we agree it is sufficient now ?